### ISSUE TITLE
[bug] chainerx.linspace

### ISSUE THREAD

**Issue by kshitij12345** [chainer/chainer#6604]
* Conditions
  * Platform: Linux-4.15.0-46-generic-x86_64-with-debian-stretch-sid
  * Chainer: 6.0.0b3
  * NumPy: 1.16.2
  * CuPy:
    * CuPy Version          : 6.0.0b3
    * CUDA Root             : /usr/local/cuda-9.2
    * CUDA Build Version    : 9020
    * CUDA Driver Version   : 10010
    * CUDA Runtime Version  : 9020
    * cuDNN Build Version   : 7500
    * cuDNN Version         : 7500
    * NCCL Build Version    : 2104
    * NCCL Runtime Version  : (unknown)
  * iDeep: Not Available

* Code to reproduce
```python
>>> import chainerx as chx
>>> import numpy as np
>>> chx_lin = chx.linspace(1,2)
>>> num_lin = np.linspace(1,2)
>>>  chx_lin
array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 2], shape=(50,), dtype=int32, device='native:0')
>>> num_lin
array([1.        , 1.02040816, 1.04081633, 1.06122449, 1.08163265,
       1.10204082, 1.12244898, 1.14285714, 1.16326531, 1.18367347,
       1.20408163, 1.2244898 , 1.24489796, 1.26530612, 1.28571429,
       1.30612245, 1.32653061, 1.34693878, 1.36734694, 1.3877551 ,
       1.40816327, 1.42857143, 1.44897959, 1.46938776, 1.48979592,
       1.51020408, 1.53061224, 1.55102041, 1.57142857, 1.59183673,
       1.6122449 , 1.63265306, 1.65306122, 1.67346939, 1.69387755,
       1.71428571, 1.73469388, 1.75510204, 1.7755102 , 1.79591837,
       1.81632653, 1.83673469, 1.85714286, 1.87755102, 1.89795918,
       1.91836735, 1.93877551, 1.95918367, 1.97959184, 2.        ])

```

**Comment by kshitij12345** [comment]
https://github.com/numpy/numpy/issues/8597

Referring to above issue, ` numpy.linspace` will always default to float dtype.

I would like to send a PR , doing the same thing here.

**Comment by hvy** [comment]
Fixed and closed by #6605 .
